src/win/pipe.c

```diff
@@ -965,9 +985,17 @@ static DWORD WINAPI uv_pipe_zero_readfile_thread_proc(void* parameter) {
     } else {
       hThread = NULL;
     }
+    DuplicateHandle(GetCurrentProcess(), GetCurrentThread(),
+                    GetCurrentProcess(), &hThreadForInterrupt,
+                    0, TRUE, DUPLICATE_SAME_ACCESS);
     uv_mutex_unlock(m);
   }
 restart_readfile:
+  if (hThreadForInterrupt) {
+    QueueUserWorkItem(uv__pipe_zero_read_interrupter,
+                      hThreadForInterrupt,
```

Comment:
It's undefined what Win32 will do with this handle after this function returns (I expect it should reuse it for the next IO call). Provision needs to be made to ensure that the work item scheduled here doesn't try to access this handle asynchronously. This is what the mutex locks in `uv__pipe_pause_read ` are required for.